pex/pip/version.py

```diff
@@ -68,7 +78,8 @@ def values(cls):
         return cls._values
 
     v20_3_4_patched = PipVersionValue(
-        version="20.3.4-patched",
+        name="20.3.4-patched",
+        version="20.3.4+patched",
```

Comment:
I didn't grok why the `name/version` distinction is necessary. For the currently supported versions at least ordering by the (pre-change) `version` seems like it would work? And I don't see anything in the versions of pip releases that would defeat it even if we added new versions in the future.

Reply:
The `20.3.4-patched` is not a valid PEP-440 version. I did not want to break the CLI value though; so split the name from the version conceptually to handle this 1 case.